src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/InterchangeHeaderTest.java

```diff
@@ -89,4 +92,28 @@ void testFromStringWithInvalidEdifactStringThrowsException() {
 
         assertEquals("Can't create InterchangeHeader from wrong value", exception.getMessage());
     }
+
+    @Test
+    void testFromStringWithNhsAckRequestedSetsTrue() {
+        final var interchangeHeader = InterchangeHeader.fromString(
+                "UNB+UNOA:2+SNDR+RECP+190323:0900+00000001++MEDRPT++1");
+
+        assertEquals(InterchangeHeader.KEY, interchangeHeader.getKey());
+        assertEquals("SNDR", interchangeHeader.getSender());
+        assertEquals("RECP", interchangeHeader.getRecipient());
+        assertEquals(1L, interchangeHeader.getSequenceNumber());
+        assertTrue(interchangeHeader.isNhsAckRequested());
```

Comment:
We can use `assertAll()` to see as many failures as possible, if something goes wrong
```suggestion
        assertAll(
                () -> assertEquals(InterchangeHeader.KEY, interchangeHeader.getKey()),
                () -> assertEquals("SNDR", interchangeHeader.getSender()),
                () -> assertEquals("RECP", interchangeHeader.getRecipient()),
                () -> assertEquals(1L, interchangeHeader.getSequenceNumber()),
                () -> assertTrue(interchangeHeader.isNhsAckRequested())
        );
```